lib/core/src/persist/mod.rs

```diff
@@ -114,7 +114,7 @@ impl Persister {
         Ok(persister)
     }
 
-    pub(crate) fn get_connection(&self) -> Result<Connection> {
+    pub fn get_connection(&self) -> Result<Connection> {
```

Comment:
How about instead of exposing the get_connection add an optional data parameter to the Persister::new_in_memory constructor? Also we can expose serialize to return the db for backup.
This way the WASM can reload the bytes and pass it there instead of using the connection directly.

Reply:
I did dislike exposing that method. Thanks for the proposal. Done in [0b031a9](https://github.com/breez/breez-sdk-liquid/pull/851/commits/0b031a9a8de9e1fd0320c73d3043543ca91c0cc1).

It' a bit annoying that the fingerprint is "needed" for getting a good unique path. The wallet provides the fingerprint method but it can only be created with a persister. I duplicated a bit of code to derive the fingerprint in the Wasm crate as I think that is the simplest solution for now.

Reply:
Also, the previous approach had a larger issue: the recovery was made after the db was initialized, meaning that we would be missing migrations. Now, we recover before applying migrations.